boot/bootutil/src/loader.c

```diff
@@ -930,16 +929,19 @@ boot_validated_swap_type(struct boot_loader_state *state,
      */
 
     if (hdr->ih_magic == IMAGE_MAGIC) {
-        vtable_addr = (uint32_t)hdr + hdr->ih_hdr_size;
-        vtable = (uint32_t *)(vtable_addr);
-        reset_addr = vtable[1];
+        rc = flash_area_read(secondary_fa, hdr->ih_hdr_size +
+				 sizeof(uint32_t), &reset_addr,
+				 sizeof(reset_addr));
+        if (rc != 0) {
+            return BOOT_SWAP_TYPE_FAIL;
+        }
 #ifdef PM_S1_ADDRESS
 #ifdef PM_CPUNET_B0N_ADDRESS
         if(reset_addr < PM_CPUNET_B0N_ADDRESS)
 #endif
         {
             const struct flash_area *primary_fa;
-            int rc = flash_area_open(flash_area_id_from_multi_image_slot(
+        rc = flash_area_open(flash_area_id_from_multi_image_slot(
```

Comment:
alignment

Reply:
@osaether can you please fix this? We are about to merge for 2.1.0 rc1